### DISCUSSION TITLE
Hook for writing to package.json file

### DISCUSSION THREAD

**Discussion by vjpr** [pnpm/pnpm#5624] [Ideas]
This is related to: https://github.com/pnpm/pnpm/issues/3027

---

I would like to use `package.json5` format. Mainly for adding comments to certain dependencies and for scripts. At the moment I use an adjacent `package.json.md` file.

---

In a large monorepo, the same fields are often repeated in every `package.json` file. Ideally, `package.json` files would be generated.

Nx already does this. It [merges](https://nx.dev/reference/project-configuration) `package.json` and `project.json` when running a task.

---

A lot of tools still rely on `package.json` such as IntelliJ. So `package.json` must always be kept up to date.

When running `pnpm add|link|remove`, I'd like it to update my `package.json5` file, but also my `package.json` file.

1. At present pnpm prioritizes `package.json` over `package.json5`. A config option should be added for this: https://github.com/pnpm/pnpm/issues/3027
2. A hook should be available to allow syncing any updates `pnpm` makes. Something like `writePackage` in `pnpmfile.js`.

## Example

```
package.json
package.json5
pnpmfile.js
```

_pnpmfile.js_

```
export function writePackageJson() {
  fs.writeFileSync('package.json', ...)
}
```

```
pnpm add lodash

> Updates package.json5
> Runs hook to update package.json
```

**Comment by weyert** [comment]
Can’t you use meta-updater for this?

**Reply by aparajita** [reply]
> Can’t you use meta-updater for this?

It looks like technically that could be done. Anyone brave enough to figure out how to do it?
